app/src/main/java/com/example/tasky/common/data/util/DataError.kt

```diff
@@ -0,0 +1,24 @@
+package com.example.tasky.common.data.util
+
+sealed interface DataError : Error {
+    sealed class Network(val code: Int, private val message: String = "") : DataError {
+        fun formatErrorMessage(): String {
+            return "${this.code}: ${this.message}"
+        }
+
+        class RequestTimeout(message: String = "Request Timeout") : Network(408, message)
+        class Unauthorized(message: String = "Unauthorized") : Network(401, message)
+        class PayloadTooLarge(message: String = "Payload Too Large") : Network(413, message)
+        class Unknown(message: String = "Unknown Error") : Network(-2, message)
+        class NoInternet(message: String = "No Internet") : Network(0, message)
```

Comment:
In the data and domain layers, you're generally fine with just knowing the type of error that happened without giving it a specific message. Deciding which error type results in which displayed error message is responsibility of the presentation layer, since that's the only place where an error message is needed, so you could just have an extension function there that parses it:
```
// In presentation
fun DataError.toErrorMessage(): String {
    // ...
}
```